framework/meta-lib/src/contract/generate_snippets/snippet_template_gen.rs

```diff
@@ -55,6 +55,15 @@ async fn main() {{
     // all contracts have a deploy snippet
     writeln!(file, r#"        "deploy" => interact.deploy().await,"#).unwrap();
 
+    for upgrade_endpoint in &abi.upgrade_constructors {
+        writeln!(
+            file,
+            r#"        "{}" => interact.{}().await,"#,
+            upgrade_endpoint.name, upgrade_endpoint.rust_method_name
```

Comment:
upgrade_endpoint.name is always "upgrade"
make it a constant somewhere (or use the one that already exists)

Reply:
In fact, there shouldn't be a `name` field there, it's redundant, since only one value is allowed